game2-safe/src/Server/Server.fs

```diff
@@ -0,0 +1,129 @@
+open System
+open System.IO
+open System.Threading.Tasks
+open System.Text
+
+open Microsoft.AspNetCore.Builder
+open Microsoft.Extensions.DependencyInjection
+open FSharp.Control.Tasks.V2
+open Giraffe
+open Saturn
+open Shared
+
+open Fable.Remoting.Server
+open Fable.Remoting.Giraffe
+open Microsoft.WindowsAzure.Storage
+open Microsoft.WindowsAzure.Storage.Table
+open Microsoft.WindowsAzure.Storage.Blob
+open Newtonsoft.Json
+open Microsoft.WindowsAzure.Storage
+open Microsoft.WindowsAzure.Storage
+
+let tryGetEnv = System.Environment.GetEnvironmentVariable >> function null | "" -> None | x -> Some x
+let publicPath = tryGetEnv "public_path" |> Option.defaultValue "../Client/public" |> Path.GetFullPath
+let storageAccount = tryGetEnv "STORAGE_CONNECTIONSTRING" |> Option.defaultValue "UseDevelopmentStorage=true" |> CloudStorageAccount.Parse
+
+let blobClient = storageAccount.CreateCloudBlobClient()
+let guid = "151be5b4-3cb7-45b7-8737-65e08e8faf63"
+let blobRef = blobClient.GetContainerReference guid
+
+let blob =
+    task {
+        let! _ = blobRef.CreateIfNotExistsAsync ()
+        do! blobRef.SetPermissionsAsync
+                (BlobContainerPermissions(
+                    PublicAccess = BlobContainerPublicAccessType.Blob
+                ))
+        let file =
+            blobRef.GetBlockBlobReference "scores.json"
+        let! exists = file.ExistsAsync()
+        if not exists then
+            do! file.UploadTextAsync "[]"
+        return file
+    }
+    |> Async.AwaitTask
+    |> Async.RunSynchronously
+
+let jsonConverter = Fable.Remoting.Json.FableJsonConverter () :> JsonConverter
+
+let leaseTime = TimeSpan.FromSeconds 15.
+
+let getScores () =
+    task {
+        let b = Array.zeroCreate 65536
+        let! count = blob.DownloadToByteArrayAsync(b, 0)
+        let json = Encoding.UTF8.GetString(b, 0, count)
+        return JsonConvert.DeserializeObject<HighScores>(json, [|jsonConverter|])
+    }
+
+let addScore (score : string * int) =
+    task {
+        let! lease = blob.AcquireLeaseAsync(Nullable.op_Implicit leaseTime, null)
+        let! scores = getScores ()
+        let scores = score :: scores
+        let json = JsonConvert.SerializeObject(scores, [|jsonConverter|])
+        let bytes = Encoding.UTF8.GetBytes json
+        let leaseCondition = AccessCondition.GenerateLeaseCondition(lease)
+        do! blob.UploadFromByteArrayAsync(
+                bytes, 0, bytes.Length,
+                leaseCondition, BlobRequestOptions(), OperationContext())
```

Comment:
The last two arguments are (AFAIK) optional and can be omitted (or can just be null).